internal/services/applications/parse/service_principal.go

```diff
@@ -68,3 +68,13 @@ func (id ServicePrincipalId) Segments() []resourceids.Segment {
 func (id ServicePrincipalId) String() string {
 	return fmt.Sprintf("ServicePrincipal (Object ID: %q)", id.ServicePrincipalId)
 }
+
+func (id ServicePrincipalId) FromParseResult(input resourceids.ParseResult) error {
```

Comment:
This'll need to be a pointer:

```suggestion
func (id *ServicePrincipalId) FromParseResult(input resourceids.ParseResult) error {
```